src/content/blog/mariadb-dump-breaking-change.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "MariaDB Dump (mysqldump) Breaking Change"
+pubDate: 2024-06-14
+#modifiedDate: 2024-04-23
+summary: What is the breaking change in Mariadb-dump (mysqldump) and what does it mean for my DDEV projects?
+author: Randy Fay
+featureImage:
+  src: /img/blog/2024/06/weeping-sea-lion.png
+  alt: "Dall-E: an image of a seal posed and crying with big tear drops, something like the MariaDB logo"
+  credit: 'Dall-E: an image of a seal posed and crying with big tear drops, something like the MariaDB logo'
+categories:
+  - DevOps
+---
+
+## What happened with MariaDB and MysqlDump?
+
+On May 17, 2024, MariaDB responded to a security issue by creating a new directive in the output of `mariadb-dump`/`mysqldump`. This new directive in the dump file looks like this:
+
+`/*!999999\- enable the sandbox mode */`
+
+This directive/content is incompatible with all versions of the `mysql` or `mariadb` database client from all vendors from before that May 17, 2024. 
+
+Trying to import a database dump created by `mariadb-dump` (usually aliased to `mysqldump`) in one of the new versions of their product results in:
+
+`ERROR at line 1: Unknown command '\-'`
+
+All currently maintained MariaDB server versions got this breaking change, including 10.5.25, 10.6.18, 10.11.8, 11.0.6, 11.1.5, 11.2.4 and 11.4.2.
+
+You can read the details in the innocently titled article [MariaDB Dump File Compatibility Change](https://mariadb.org/mariadb-dump-file-compatibility-change/)
+
+## What does it mean To my DDEV projects?
+
+### DDEV v1.23.1
+
+If you are using DDEV v1.23.1, we mitigated this problem at some levels by updating the DDEV v1.23.1 `ddev-dbserver` image. This solved a number of problems related to `ddev import-db` and `ddev export-db` because the format used was the new (breaking-change) MariaDB format. If you see this on `ddev import-db` or `ddev export-db` you can update to the newer version using the appropriate command below:
```

Comment:
```suggestion
If you are already on DDEV v1.23.1, we've mitigated this problem to a certain extent by updating the `ddev-dbserver` image for this release. This solved a number of problems related to `ddev import-db` and `ddev export-db` because the format used was the new MariaDB format containing the breaking change. If you see this on `ddev import-db` or `ddev export-db` you can update to the newer version using the appropriate command below:
```

And what does "if you see this" refer to? to me it is not clear at all. i thought the snippet underneath are the instructions for updating to the newer version?

Reply:
> And what does "if you see this" refer to?

If you see the disaster in importing.